src/Mod/PartDesign/Gui/Command.cpp

```diff
@@ -1048,15 +1049,22 @@ void prepareProfileBased(PartDesign::Body *pcActiveBody, Gui::Command* cmd, cons
                 }
             }
         }
-        else {
-            if (feature->isDerivedFrom(Part::Part2DObject::getClassTypeId()) || subs.empty())
+        else if (which.compare("AdditivePipe") == 0 ||
+                 which.compare("SubtractivePipe") == 0) {
+            // for additive and subtractive pipes set subvalues even for sketches
+            // to support point sections
+            auto subName = subs.empty() ? "" : subs.front();
+
+            // `ProfileBased::getProfileShape()` and other methods will return
+            // just the sub-shapes if they are set. So when whole sketches are
+            // desired, don not set sub-values.
+            if (feature->isDerivedFrom(Part::Part2DObject::getClassTypeId()) &&
+                !(subName.size() > 6 && subName.substr(0,6) == "Vertex"))
```

Comment:
Where does the 6 comes from? Can you please add a comment?

nit-pick: our style is to have a space after commas.

Reply:
6 is for the characters in "Vertex". A snippet copied from elsewhere in PD, which is why the space-less comma.

Reply:
You use the same code for lofts.

But is this check safe? What subNames could we have?

Reply:
From https://www.cplusplus.com/reference/string/string/substr/:

> len
>    Number of characters to include in the substring (if the string is shorter, as many characters as possible are used).

So we probably don't need the length check.

Looking at that site, I see a clearer (and probably faster and safer) method https://www.cplusplus.com/reference/string/string/compare/.

As for what other subNames we can have, it includes "Edge<number>" and "Face<number>". I expect after topo-naming is merged the <number> part will be replaced by a description of where those entities are within the solid but the first part remain the same. If that's not the case, we'll have to address it in the future.

Reply:
Since you also check for "Vertex", you can omit the length check I think.
Please decide.

From my point of view it is then ready to be merged.

Reply:
So this is what I'll do:
1. In `FeaturePipe` the comparison only relates to pipe, so I'll change to using `compare`
2. In `Command.cpp` and other places where there are other occasions where this snippet is used, I'll instead make a separate PR.

Does that sound good?

Reply:
Very good.

Reply:
See https://github.com/FreeCAD/FreeCAD/pull/5208.